src/DotPulsar/Internal/ConsumerChannelFactory.cs

```diff
@@ -70,6 +70,23 @@ public sealed class ConsumerChannelFactory : IConsumerChannelFactory
             }
         }
 
+        private static string GenerateRandomConsumerName()
+        {
+            try
+            {
+                using (var rng = System.Security.Cryptography.RandomNumberGenerator.Create())
+                {
+                    var bytes = new byte[4];
```

Comment:
what will happen if we have 2 consumers with the same name?

Reply:
Not sure, but we have a ton of them today (they are all named null)

Reply:
The java client autogenerates names in similar fashion, except it performs a SHA1 of a new GUID and only takes 5 characters, so ours has fewer collisions at least?

Reply:
https://github.com/apache/pulsar/blob/82eae5dc446b3167e37b67db908b909d44d1016d/pulsar-client/src/main/java/org/apache/pulsar/client/util/ConsumerName.java

Reply:
hm. well I assume null means server should assign name. I am okay as is. Might be good idea to test with 2 equal names though.